app/controllers/api/stories_controller.rb

```diff
@@ -11,16 +11,26 @@ def random
     show
   end
 
-  def resource
-    @story ||= if params[:id]
-                 Story.published.visible.find(params[:id])
-               elsif request.put? || request.post?
-                 Story.new
-               end
+  private
+
+  def resources_base
+    if highlighted?
+      account.portfolio_stories
+    else
+      super
+    end
   end
 
-  def resources
-    @stories ||= resources_base.includes(
+  def filtered(resources)
+    if highlighted?
```

Comment:
I'm confused why this `#highlighted?` check is necessary in both `#resources` and `#resources_base` - is this redundant, unnecessary, and redundant?